### PR TITLE
Issue / .NET 8 / Default Lambda Parameters

### PR DESCRIPTION
**Pull request by alptuncs**
Learn how to use default lambda parameters

## Tasks

- [x] Read documentation and learn
- [x] Write documentation
- [x] Add example usage to project

### CODE REVIEW

File: migration-plan.md [hunk 1]
```diff
@@ -92,7 +92,7 @@ int[] anotherArray = [.. array, .. list];
 
 ## Default lambda parameters
 
-You can now define default values for parameters on lambda expressions.
+You can now define default values for parameters on lambda expressions. .
```
Comment (SeferMirza) [inline migration-plan.md:95]: The dot at the end seemed unnecessary to me

File: chsarp/DefaultLambdaParameters.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        };
+
+        incrementValue();
+        incrementValue(5);
+
+        var action = incrementValue;
+        action();
+    }
+
+    public void ParamsArrayParameters()
```
Comment (SeferMirza) [inline chsarp/DefaultLambdaParameters.cs:23]: Is this a method related to the class it is in? It feels like the class name is different with what it does.

File: chsarp/DefaultLambdaParameters.cs [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+            _logger.LogInformation($"{(value == 2 ? "Called with default value" : "Called with given value")}: {value}");
+
+            return value + 1;
+        };
+
+        incrementValue();
+        incrementValue(5);
+
+        var action = incrementValue;
+        action();
```
Comment (SeferMirza) [inline chsarp/DefaultLambdaParameters.cs:20]: I think it's unnecessary since you're already calling twice above.

Comment (cihandeniz) [reply chsarp/DefaultLambdaParameters.cs:20]: i agree. if it showcases something else, you need to be clear about it.

Comment (alptuncs) [reply chsarp/DefaultLambdaParameters.cs:20]: This is to showcase the new behaviour which previously gave error due to a mismatch in the number of expected arguments. https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-12.0/lambda-method-group-defaults#current-behavior

Comment (cihandeniz) [reply chsarp/DefaultLambdaParameters.cs:20]: nice :ok_hand: 

then don't imply it, be explicit. you can rename `action` to something else to be explicit.

File: chsarp/DefaultLambdaParameters.cs [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+public class DefaultLambdaParameters(ILogger<CollectionExpressions> _logger)
+{
+    public void DefaultParameters()
+    {
+        var incrementValue = (int value = 2) =>
+        {
+            _logger.LogInformation($"{(value == 2 ? "Called with default value" : "Called with given value")}: {value}");
+
+            return value + 1;
+        };
```
[the remark this answers is not in the document]
Comment (cihandeniz) [reply chsarp/DefaultLambdaParameters.cs:14]: i see you've followed microsoft's samples. but they included the expected result in comments, e.g., `// 3`. if you add comments, that's fine as well.

**Review by cihandeniz — APPROVED** [review]
